diagnostic.md

```diff
@@ -3,36 +3,42 @@
 What are three advantages of using Ember?
 
 ```sh
-# your answer here
+- Each view state has its own URL - managed by routes
+- Writing less AJAX calls.
+- data display in the front end is faster due to fewer ajax calls.
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
 ```sh
-# your answer here
+components.
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
 ```sh
-# your answer here
+Routes and models
 ```
 
 How does a component differ from an Ember template?
 
 ```sh
-# your answer here
+A component is invoked from within a template. Its scope is defined at the
+time it is invoked.
 ```
 
 What is an opinionated framework? Is Ember opinionated? How or how not?
 
 ```sh
-# your answer here
+ember is an opinionated framework in that it forces you to structure your app
+in the way it wnats you to.
```

Comment:
👍 
An opinionated framework favors convention over configuration. Ember is an opinionated
framework. It requires users to name files in certain ways, and use it's actions
and data structures in certain ways.